gmapping/src/slam_gmapping.cpp

```diff
@@ -152,6 +152,9 @@ void SlamGMapping::init()
 {
   // log4cxx::Logger::getLogger(ROSCONSOLE_DEFAULT_NAME)->setLevel(ros::console::g_level_lookup[ros::console::levels::Debug]);
 
+  got_first_scan_ = false;
+  got_map_ = false;
```

Comment:
I moved this initialization to the header.

https://github.com/allenh1/slam_gmapping/blob/ros2_port/gmapping/src/slam_gmapping.hpp#L108-L109